### ISSUE TITLE
Respect of the formatting of the template (line breaks, spacing...)

### ISSUE THREAD

**Issue by FALLAI-Denis** [johnknoop/vscode-handlebars-preview#29]
We plan to use Handlebars to generate program source codes from template, (so template and result are plain text, html).
The preview should respect the formatting of the template:
- on line breaks
- on spacings

Also see if it would be possible to save the preview to keep the result.

See simulation of the expected result by adding <pre> </pre> tags to the template:
![hbs_preview_001](https://user-images.githubusercontent.com/63861965/106887660-c09d2480-66e5-11eb-9e86-a40c417d548d.png)

**Comment by utillity** [comment]
yeah, would be great! Other preview engines detect type or go by extended file-extension (ie. .html.hbs/ .md.hbs. etc)

**Comment by johnknoop** [comment]
Hi @FALLAI-Denis and sorry for the late response. I didn't see this issue until I got an e-mail notification yesterday when @utillity commented on it.

The extension is using a `createWebviewPanel` to render the preview, which assumes that the content is HTML (which is why all white space get collapsed).

I suppose it would be possible to wrap a `pre` tag around the file contents, _IF_ we can safely determine that the file is not intended to be rendered as HTML. Perhaps the _MyTemplate.csharp.hbs_ strategy could be one way, although I'm not sure I can come up with a bulletproof way to determine that a template is meant to be rendered as plain text by looking at the filename alone. For example, what if someone has a HTML template named "my.little.page.about.csharp.hbs". It would be a breaking change if we suddenly stopped rendering that as HTML. But I'm open to suggestions if you can formalize a rule that doesn't cause problems for existing users.

**Comment by utillity** [comment]
Good point! I would say just make is a setting to "auto-detect" based on filename, or based on first 50 characters containing "<html" or something

**Comment by FALLAI-Denis** [comment]
Thanks @johnknoop 

The rendering could be determined by the content after transformation:
- if the first text is `<html>` then render html as a web page
- otherwise rendered plain text "as is"

The rendering could also be determined by the command activated to produce the transform:
- preview in html format
- preview in plain text format
This second solution is therefore not linked to the content, but to the choice of the user.

**Comment by utillity** [comment]
good point - I like latter one the most :)

**Comment by angularsen** [comment]
Bump for visibility 😊

My thoughts:
- Add separate commands for HTML and plain text, would probably cover most needs.
- Nice to have: Main command auto-detects html, markdown, etc.
- Idea: Not sure how, but another plugin I tested a long time ago for Scriban/Liquid showed the raw rendered text (HTML) in one panel, then I could use any plugin of choice to open a preview of that panel. So 3 panels, template => rendered text/HTML => preview of HTML. It seemed like a flexible approach, but the disadvantage was that it required 3 panels instead of just 2.
